Release notes for 2023-06-06 (#2017)
microsoft/typespec: docs/release-notes/release-2023-06-06.md

---
title: June 2023
---

# Release Notes June 2023 (2023-06-06)

:::danger
This release contains **breaking changes**
:::

## New Features

- New decimal types

> - Support for documentation comments

Reviewer: ```suggestion
- Support for using doc comments instead of the @doc decorator
```

(I'm not even sure Documentation Comments are the same thing as doc comments).

- New `valueof` type constraint modifier for type literals
- Lazy evaluation of emitter configuration
- Signature Help for Template Arguments and other IDE improvements

> - Json Schema Emitter

Reviewer: I feel like a new emitter type should get top billing? 😂

### New Decimal Types

Added support for `decimal` and `decimal128` types

```typespec
model Foo {
  price: decimal;
  shortPrice: decimal128;
}
```

> 

Reviewer: ```suggestion
These types represent values stored in decimal format which are often used in financial calculations to avoid rounding errors introduced by binary floating point formats. `decimal` can store values of any size and precision, while `decimal128` corresponds to the 128 bit decimal floating point format described by IEEE 754.
```

### Support for Doc Comments

Doc comments (/\*\* \*/) will be treated as the documentation for types, unless there is an explicit `@doc` decorator

```typespec
/** This will now be the documentation for model Foo */
model Foo {
  name: string;
}
```

### New `valueof` type constraint modifier to require type literals

> Templates and decorators can now contrain value to literal types. For example, the following template constraint:

Reviewer: ```suggestion
Templates and decorators can now constrain arguments to literal types. For example, the following template constraint:
```

```typespec
model Foo<T extends valueof string> {
  @doc(T)
  name: string;
}

> ```

Author: I have that one under breakign changes

### Support for Doc Comments

> Doc comments (/\*\* \*/) will be treated as the documentation for types, unless there is an explicit `@doc` decorator

Reviewer 2: ```suggestion
Doc comments (`/** */`) will be treated as the documentation for types, unless there is an explicit `@doc` decorator
```
would escaping in code block be better here?
